predicators/nsrt_learning/strips_learning/effect_search_learner.py

```diff
@@ -329,77 +327,64 @@ def get_successors(
                                              get_successors=get_successors,
                                              heuristic=heuristic)
 
-        # Extract the best effect set.
-        best_effect_sets = path[-1]
-        # Convert into PNADs.
-        final_pnads = self._effect_sets_to_pnads(best_effect_sets)
-        return final_pnads
+        # Extract the best PNADs set.
+        final_pnads = path[-1]
+        # Recompute these PNADs so that they exactly match the PNADs used
+        # to compute the final heuristic.
+        recomp_final_pnads = self._recompute_pnads_from_effects(
+            list(final_pnads))
+        # Fix naming.
```

Comment:
can we move this part to a parent class too? it's used elsewhere right?

Reply:
no - it's actually not used in the `BackchainingSTRIPSLearner` at all. This is a unique feature of our new learner!